plugins/woocommerce/src/StoreApi/Routes/V1/Agentic/CheckoutSessions.php

```diff
@@ -0,0 +1,452 @@
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Set buyer data on customer.
+	 *
+	 * @param array $buyer Buyer data.
+	 */
+	protected function set_buyer_data( $buyer ) {
+		$customer = WC()->customer;
+
+		if ( isset( $buyer['first_name'] ) ) {
+			$first_name = wc_clean( wp_unslash( $buyer['first_name'] ) );
+			$customer->set_billing_first_name( $first_name );
+			$customer->set_shipping_first_name( $first_name );
+		}
+
+		if ( isset( $buyer['last_name'] ) ) {
+			$last_name = wc_clean( wp_unslash( $buyer['last_name'] ) );
+			$customer->set_billing_last_name( $last_name );
+			$customer->set_shipping_last_name( $last_name );
+		}
+
+		if ( isset( $buyer['email'] ) ) {
+			$email = sanitize_email( wp_unslash( $buyer['email'] ) );
+			if ( is_email( $email ) ) {
+				$customer->set_billing_email( $email );
+			}
+		}
+
+		if ( isset( $buyer['phone_number'] ) ) {
+			$phone = wc_clean( wp_unslash( $buyer['phone_number'] ) );
+			$customer->set_billing_phone( $phone );
+		}
+
+		$customer->save();
+	}
+
+	/**
+	 * Set fulfillment address on customer.
+	 *
+	 * @param array $address Address data.
+	 */
+	protected function set_fulfillment_address( $address ) {
+		$customer = WC()->customer;
+
+		// Parse name into first and last.
+		$name       = isset( $address['name'] ) ? wc_clean( wp_unslash( $address['name'] ) ) : '';
+		$name_parts = $name ? explode( ' ', $name, 2 ) : array( '', '' );
+		$first_name = $name_parts[0];
+		$last_name  = isset( $name_parts[1] ) ? $name_parts[1] : '';
+
+		// Sanitize all address fields.
+		$line_one    = wc_clean( wp_unslash( $address['line_one'] ?? '' ) );
+		$line_two    = wc_clean( wp_unslash( $address['line_two'] ?? '' ) );
+		$city        = wc_clean( wp_unslash( $address['city'] ?? '' ) );
+		$state       = wc_clean( wp_unslash( $address['state'] ?? '' ) );
+		$postal_code = wc_clean( wp_unslash( $address['postal_code'] ?? '' ) );
+		$country     = wc_clean( wp_unslash( $address['country'] ?? '' ) );
+
+		// Set shipping address.
+		$customer->set_shipping_first_name( $first_name );
+		$customer->set_shipping_last_name( $last_name );
+		$customer->set_shipping_address_1( $line_one );
+		$customer->set_shipping_address_2( $line_two );
+		$customer->set_shipping_city( $city );
+		$customer->set_shipping_state( $state );
+		$customer->set_shipping_postcode( $postal_code );
+		$customer->set_shipping_country( $country );
+
+		// Also set as billing address if not already set.
+		if ( ! $customer->get_billing_address_1() ) {
+			$customer->set_billing_first_name( $first_name );
+			$customer->set_billing_last_name( $last_name );
+			$customer->set_billing_address_1( $line_one );
+			$customer->set_billing_address_2( $line_two );
+			$customer->set_billing_city( $city );
+			$customer->set_billing_state( $state );
+			$customer->set_billing_postcode( $postal_code );
+			$customer->set_billing_country( $country );
+		}
+
+		$customer->save();
+
+		// Recalculate shipping.
+		WC()->cart->calculate_shipping();
```

Comment:
I wonder if this should be moved outside of the set/clear fulfilment methods, so it is in one place only.